src/main/scala/utils/images/Image.scala

```diff
@@ -251,8 +274,36 @@ trait VectorizedImage extends Image {
   override def put(x: Int, y: Int, channelIdx: Int, newVal: Double) = {
     putInVector(imageToVectorCoords(x, y, channelIdx), newVal)
   }
+
+  def vectorToImageCoords(v: Int): Coordinate = ???
+
+  val coord: Coordinate = new Coordinate(0,0,0)
+
+  def iter(): Iterator[CoordinateValue] = new Iterator[CoordinateValue] {
```

Comment:
Would be good to document that the reference to CoordinateValue shouldn't be modified or saved by the caller.